Frameworks/include/CGImageInternal.h

```diff
@@ -96,31 +106,274 @@ typedef enum {
+        const __CGImagePixelProperties* properties = Properties();
+        RETURN_RESULT_IF_NULL(properties, 0);
+        return properties->bitsPerComponent;
+    }
+
+    inline CGBitmapInfo BitmapInfo() const {
+        const __CGImagePixelProperties* properties = Properties();
+        RETURN_RESULT_IF_NULL(properties, 0);
+        return properties->bitmapInfo;
+    }
+
+    inline CGImageAlphaInfo AlphaInfo() const {
+        return static_cast<CGImageAlphaInfo>(BitmapInfo() & kCGBitmapAlphaInfoMask);
+    }
+
+    inline CGColorSpaceRef ColorSpace() {
+        const __CGImagePixelProperties* properties = Properties();
+        RETURN_NULL_IF(!properties);
+        return _CGColorSpaceCreate(properties->colorSpaceModel);
+    }
+
+    inline void SetImageSource(Microsoft::WRL::ComPtr<IWICBitmapSource> source) {
+        bitmapImageSource = std::move(source);
+        // populate the image info.
+        if (FAILED(bitmapImageSource->GetSize(&width, &height))) {
+            height = 0;
+            width = 0;
+        }
+
+        bitmapInfo = BitmapInfo();
+        alphaInfo = AlphaInfo();
+        bitsPerPixel = BitsPerPixel();
+        bitsPerComponent = BitsPerComponent();
+        bytesPerRow = (bitsPerPixel >> 3) * width;
+        if (!colorSpace) {
+            colorSpace.reset(ColorSpace());
+        }
+    }
+};
+
+struct __CGImage : CoreFoundation::CppBase<__CGImage, __CGImageImpl> {
+    inline Microsoft::WRL::ComPtr<IWICBitmapSource>& ImageSource() {
+        return _impl.bitmapImageSource;
+    }
+
+    inline size_t Height() const {
+        return _impl.height;
+    }
+
+    inline size_t Width() const {
+        return _impl.width;
+    }
+
+    inline bool IsMask() const {
+        return _impl.isMask;
+    }
+
+    inline bool Interpolate() const {
+        return _impl.interpolate;
+    }
+
+    inline CGColorSpaceRef ColorSpace() {
+        return _impl.colorSpace.get();
+    }
+
+    inline CGColorRenderingIntent RenderingIntent() const {
+        return _impl.renderingIntent;
+    }
+
+    inline CGBitmapInfo BitmapInfo() const {
+        return _impl.bitmapInfo;
+    }
+
+    inline CGImageAlphaInfo AlphaInfo() const {
+        return _impl.alphaInfo;
+    }
+
+    inline size_t BitsPerPixel() const {
+        return _impl.bitsPerPixel;
+    }
+
+    inline size_t BytesPerRow() const {
+        return _impl.bytesPerRow;
+    }
+
+    inline size_t BitsPerComponent() const {
+        return _impl.bitsPerComponent;
+    }
+
+    inline __CGImage& SetImageSource(Microsoft::WRL::ComPtr<IWICBitmapSource> source) {
+        _impl.SetImageSource(source);
+        return *this;
+    }
+
+    inline __CGImage& SetIsMask(bool mask) {
+        _impl.isMask = mask;
+        return *this;
+    }
+
+    inline __CGImage& SetInterpolate(bool interpolate) {
+        _impl.interpolate = interpolate;
+        return *this;
+    }
+
+    inline __CGImage& SetColorSpace(CGColorSpaceRef space) {
```

Comment:
Already addressed in earlier comment, please take a look.

---
In reply to: [85251955](https://github.com/Microsoft/WinObjC/pull/1185#discussion_r85251955) [](ancestors = 85251955)